ArduPlane/parachute.cpp

```diff
@@ -56,3 +59,62 @@ bool Plane::parachute_manual_release()
 
     return true;
 }
+
+// Code to detect loss of control for ArduPlane
+#define PARACHUTE_CHECK_TRIGGER_SEC         1  // 1 second of loss of control triggers the parachute
+
+/*
+  parachute_emergency_check - trigger the release of the parachute
+  automatically if a critical situation is detected
+*/
+void Plane::parachute_emergency_check()
+{
+    static uint32_t control_loss_ms;  // time when we first lost control and never regained it
```

Comment:
There is no "plan" but the concern is a relatively new one. Since ArduPilot is moving more and more towards Linux based systems you never know how the class ends up getting used! Not using static variables removes some chances of bugs. If things were to change it's easier to enforce it during development instead of all at once at the end. The other uses of static have been there a long time and should be changed too.